debezium-core/src/main/java/io/debezium/pipeline/ChangeEventSourceCoordinator.java

```diff
@@ -84,6 +86,16 @@ public ChangeEventSourceCoordinator(O previousOffset, ErrorHandler errorHandler,
 
     public synchronized void start(CdcSourceTaskContext taskContext, ChangeEventQueueMetrics changeEventQueueMetrics,
                                    EventMetadataProvider metadataProvider) {
+        if (previousOffsets.size() != 1) {
+            throw new ConnectException("The coordinator must be provided with exactly one partition, "
+                    + previousOffsets.size() + " found");
+        }
+
+        Map.Entry<P, O> entry = previousOffsets.entrySet().iterator().next();
+
+        final P partition = entry.getKey();
```

Comment:
So this is working with a single partition here; is this only temporary until further work in DBZ-2975, and `ChangeEventSourceCoordinator` would eventually deal with multiple partitions (if supported by a connector)? Or will `ChangeEventSourceCoordinator` always stay single-partition scoped and iteration over partitions will happen elsewhere?

Reply:
> So this is working with a single partition here; is this only temporary until further work in DBZ-2975, and `ChangeEventSourceCoordinator` would eventually deal with multiple partitions (if supported by a connector)?

Yes, this is just to adapt the existing single-partition implementations to the new multi-partition API being introduced.

> Or will ChangeEventSourceCoordinator always stay single-partition scoped and iteration over partitions will happen elsewhere?

No, it will become partition-aware as well. It will iterate over partitions and query the sources with each of them.